File: benchmarks/python/11_string_concat_scan.py
def main():
    parts = []
    i = 0
    while i < 250:
        parts.append("item=" + str(i) + ";")
        i += 1

    text = "".join(parts)
    digits = 0
    equals = 0
    semicolons = 0
    for ch in text:
        if "0" <= ch <= "9":
            digits += 1
        elif ch == "=":
            equals += 1
        elif ch == ";":
            semicolons += 1

    print(len(text) + digits + equals + semicolons)


if __name__ == "__main__":
    main()

File: benchmarks/c/11_string_concat_scan.c
// String Concat and Scan Benchmark - C Implementation

#include <stdio.h>
#include <stdlib.h>

int main() {
    const int n = 250;
    const size_t capacity = 4096;
    char* text = (char*)malloc(capacity);
    size_t len = 0;

    for (int i = 0; i < n; i++) {
        len += (size_t)snprintf(text + len, capacity - len, "item=%d;", i);
    }

    long digits = 0;
    long equals = 0;
    long semicolons = 0;
    for (size_t i = 0; i < len; i++) {
        char ch = text[i];
        if (ch >= '0' && ch <= '9') {
            digits++;
        } else if (ch == '=') {
            equals++;
        } else if (ch == ';') {
            semicolons++;
        }
    }

    printf("string concat checksum: %ld\n", (long)len + digits + equals + semicolons);
    free(text);
    return 0;
}
